### ISSUE TITLE
LEGO® The Lord of the Rings (214510)

### ISSUE THREAD

**Issue by danieldukic** [ValveSoftware/Proton#1836]
# Compatibility Report
- Name of the game with compatibility issues: LEGO® The Lord of the Rings
- Steam AppID of the game: 214510

## System Information
- GPU: Intel® HD Graphics 4400
- Driver/LLVM version: 3.0 Mesa 18.3.0-devel - padoka PPA
- Kernel version: 4.15.0-38-generic
- Link to full system information report as [Gist](https://gist.github.com/danieldukic/7ed8696ce6a73156adfb7698651853a3):
- Proton version: 3.7-8

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
The game launches, shows attached screen and then crashes.
![lego](https://user-images.githubusercontent.com/31858712/47613886-4063cf80-da97-11e8-8c64-02701032b419.png)

## Reproduction
Start the game.

**Comment by kisak-valve** [comment]
Hello @danieldukic, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by danieldukic** [comment]
@kisak-valve  [steam-214510.log](https://github.com/ValveSoftware/Proton/files/2522803/steam-214510.log)

**Comment by legluondunet** [comment]
Same result for me: You launch the game, the logo screen appeared then the game crashed.
I joined you the log.
[steam-214510.log](https://github.com/ValveSoftware/Proton/files/2700774/steam-214510.log)

**Comment by sprite-1** [comment]
```WINEPREFIX=~/.steam/steam/steamapps/compatdata/214510/pfx/ winetricks directx9```

**Comment by lucifertdark** [comment]
> `WINEPREFIX=~/.steam/steam/steamapps/compatdata/214510/pfx/ winetricks directx9`

That command is very likely to make things worse as it will use whichever standard version of Wine you have installed outside Proton, Wine4.0 is not the same as Wine 3.16.

**Comment by sprite-1** [comment]
Fair enough, but when shit goes wrong, they can just delete the `214510` folder inside `compatdata` :thinking:

**Comment by lucifertdark** [comment]
> Fair enough, but when shit goes wrong, they can just delete the `214510` folder inside `compatdata` thinking

yep that's what I had to do twice.  Got it working now though, performance isn't great but it's at least playable.

**Comment by kisak-valve** [comment]
`Internal error (please report) ../../../../proton/wine/libs/wpp/ppy.y 387: Invalid type indicator (0x0001)` looks like a line of interest in the logs.

**Comment by nmiell** [comment]
https://github.com/wine-mirror/wine/blob/master/libs/wpp/ppy.y#L96

0x0001 is SIZE_CHAR

**Comment by nmiell** [comment]
Note that it never creates expressions with type cv_uchar or cv_schar, which suggests memory corruption or something.

**Comment by nmiell** [comment]
Further observations: the game requires D3DX9_41.DLL but doesn't do anything to install it, so it ends up using Wine's implementation which is incompatible.

The question is how did it ever work in the first place on Windows? Purely by coincidence based on the user's likelihood of having other Steam games installed which already installed the necessary dependencies?

This might be fixed by adding in the "228990 DirectX Jun 2010 Redist" depot to the packages, but that probably needs cooperation from Traveller's Tales and/or Warner Interactive.

**Comment by yajo** [comment]
So did anyone make this work?

**Comment by aanderse** [comment]
@Yajo yeah I followed some instructions from https://www.protondb.com/app/214510 and it works great.

> The game initially crashed after showing the first loading screen, but it was fine after I ran the command "protontricks 214510 d3dx9_41"

**Comment by adrianlshaw** [comment]
Crashes for me on Ubuntu 18.04
Hardware: https://gist.github.com/adrianlshaw/09a3cd8be69843a6d523fb37e3b64da5
Log: https://gist.github.com/adrianlshaw/e288035037dc05337a0e7afe99689026

Same with proton-4.11-12 and 5.0

**Comment by Leopard1907** [comment]
There are multiple problems with this game. 

1-) ~~Game menu music doesn't work in Proton Experimental while it works ( menu music ) with Proton 5.13 and 5.10. So there is a regression.~~ Nvm , it is completely random. Not sure if it is an actual issue now.
2-) Game doesn't save resolution and vsync changes. For a brief moment image goes sharper and when you hit apply , it just reverts all settings back. Tried with no esync and fsync also but no luck. 

Changes made into `Steam/steamapps/compatdata/214510/pfx/drive_c/users/steamuser/Application Data/Warner Bros. Interactive Entertainment/LEGO The Lord of the Rings/pcconfig.txt` is a good workaround for this issue. They seem to be applied and does stick.

3-) Game needs `d3dx9_41` applied otherwise it crashes. Luckily protontricks covers that.

**Comment by kisak-valve** [comment]
Steam Link on NVIDIA Shield TV - Lego The Lord of the Rings - stutter when destroying things

Issue transferred from https://github.com/ValveSoftware/steam-for-linux/issues/7695.
@mss555 posted on 2021-03-07T12:26:54:

#### Your system information

* Steam link version (build number or date):  1.1.73.179
* Distribution (e.g. Ubuntu): Ubuntu 20.04.2, Mesa 20.3.4 - kisak-mesa PPA
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

First of all, a *huge* thank you for your work on Proton, making it possible to play games on Linux! Second, thank you very much for supporting Remote Play from Linux hosts supporting AMD Hardware!

I've recorded a video illustrating the issue with debug information turned on at https://youtu.be/6Aw1l0pAXgI

I've tried several client setups using the exact same host setup:

* Streaming to another Ubuntu 20.04.2 Machine using the regular Steam client: no stuttering, steady 60 FPS
* Streaming to a Raspberry Pi 4 using the Steam Link app: no stuttering, steady 60 FPS
* Playing the game locally on the host: no stuttering, steady 60 FPS

The stuttering only happens when streaming to the Steam Link app running on a NVIDIA Shield TV (I have two of them, a regular Shield TV and a Pro version and both show the same problem). I'm playing many LEGO and other Games using this Remote Play setup and so far it is only in the LotR one where I've observed these drops below 20 FPS when destroying things.

I've also played extensively with changing quality and the streaming bandwidth settings. The stuttering remains the same in all cases.

While the video shows a sequence about 10% into the game, the issue manifests right away. Just start the game and start walking around in the Shire destroying things and you'll observe the stuttering when using Remote Play via Steam Link to a NVIDIA Shield TV.

Here are my settings (also noted in the video description):

Setup Host:

Ubuntu 20.04.2
Sapphire Pulse Radeon RX 5700 (8GB)
Kernel 5.8.0-44
Driver: 4.6 (Compatibility Profile) Mesa 20.3.4 - kisak-mesa PPA
Proton 5.13-6
To be able to run Lego LotR: `protontricks 214510 d3dx9_41`
CPU: i7-6700

Setup Client:

Nvidia Shield TV Version 8.2.2 (32.6.509.0)
Steam Link Version: 1.1.73.179

Here are the requested log files:
[remote_connections.txt](https://github.com/ValveSoftware/steam-for-linux/files/6097066/remote_connections.txt)
[streaming_log.txt](https://github.com/ValveSoftware/steam-for-linux/files/6097067/streaming_log.txt)

#### Steps for reproducing this issue:

1. Install Lego LotR using `protontricks 214510 d3dx9_41`.
2. Connect via Steam Link on the NVIDIA Shield TV.
3. Start a new game which will bring you drop you in the Shire. Start destroying random things to experience the FPS drops.

**Comment by stephensrmmartin** [comment]
Hit a bug on latest proton experimental. I have the fix installed mentioned here, and it has worked great for about 15 hrs of gameplay. There is one part of the game in the 2 towers section, Warg attack, at the end, where it reliably freezes. The fix is to disable esync and fsync if they are enabled in your proton. Not sure which is responsible, but we hit the freeze at the same time, 5 times in a row, after killing two Warg riders, where only audio keeps playing. I disabled esync/fsync, and did not hit the freeze thereafter.

**Comment by TyroneWatermelon420** [comment]
got
Internal error (please report) ./dlls/d3dcompiler_43/ppy.y 369: Invalid type indicator (0x12886770)
pid 20171 != 20170, skipping destruction (fork without exec?)
Proton Experimental, 7.0-3, 6.3-8 and 411-13 tested

**Comment by berksudan** [comment]
I did the following things on my Ubuntu 22.04 LTS and the problem solved. Though I'm not really sure which step(s) solved the issue exactly. The actions I have done are here:

+ I installed `winetricks` by running this on Linux Terminal: `sudo apt install winetricks`.
+ I installed `protontricks` by running this on Linux Terminal: `sudo pip3 install protontricks`.
+ As suggested in [here](https://www.protondb.com/app/214510), I executed this on Linux Terminal: `protontricks 214510 d3dx9_41`.
+ On Steam > LEGO LOTR > Properties > General, I added this to "Launch Options": `PROTON_USE_WINED3D=1 %command%`.
+  On Steam > LEGO LOTR > Properties > Compatibility, I switched to `Proton 5.13-6`.

**Comment by kisak-valve** [comment]
(214510) LEGO The Lord of the Rings does not start

Issue transferred from https://github.com/ValveSoftware/steam-for-linux/issues/9070.
@thackel posted on 2023-01-05T14:38:17:

#### Your system information

* Steam client version (build number or date): 
Latest beta
* Distribution (e.g. Ubuntu): 
Arch
* Opted into Steam client beta?: [Yes/No]
Yes
* Have you checked for system updates?: [Yes/No]
Yes

#### Please describe your issue in as much detail as possible:

When starting the game the screen goes black and shows a copyright notice. After about 3s it crashes back to desktop.

The last relevant message shown is:
`Internal error (please report) ./dlls/d3dcompiler_43/ppy.y 369: Invalid type indicator (0x0001)`

The system is using an Intel on board graphics card.

I tried proton 7.0.5, experimental and 6.x.

I've attached to proton logfiles of the runs with 7.0.5 and experimental:
[steam-214510-proton-experimental.log](https://github.com/ValveSoftware/steam-for-linux/files/10352936/steam-214510-proton-experimental.log)
[steam-214510-proton-705.log](https://github.com/ValveSoftware/steam-for-linux/files/10352938/steam-214510-proton-705.log)

**Comment by thackel** [comment]
I confirm that [installing](https://github.com/ValveSoftware/Proton/issues/1836#issuecomment-1319131816) (thanks @berksudan) `winetricks`, `protontricks` and then `DirectX9.41` made the game working.

**Comment by waspennator** [comment]
Game boots up fine on my deck, however it seems to have an issue where the water is all rainbow seizurey. It can be seen as soon as you get to the title screen and goes away if you use protontricks to install d3dx9_41 or use proton ge

[steam-214510.log.txt](https://github.com/ValveSoftware/Proton/files/14729217/steam-214510.log.txt)

**Comment by ranplayer** [comment]
I can confirm the water misrendering with Proton 9.0-3. The suggested installation of `d3dx9_41.dll`  works.

![lego_lotr_menu](https://github.com/user-attachments/assets/e0d33c17-7f9b-41db-ae96-c3bb6d93feaa)

Logs: 
[steam-214510-proton903.tar.gz](https://github.com/user-attachments/files/17194819/steam-214510-proton903.tar.gz)

Main error lines:
```
6309.186:0128:0160:err:d3dcompiler:preprocess_shader Shader log:
6309.186:0128:0160:err:d3dcompiler:preprocess_shader     <anonymous>:270:1: W4300: Redefinition of varying_uvSet0_source.
6309.186:0128:0160:err:d3dcompiler:preprocess_shader     <anonymous>:275:1: W4300: Redefinition of FUNC_UVSET_1.
6309.186:0128:0160:err:d3dcompiler:preprocess_shader     <anonymous>:1040:2: W4300: Redefinition of STATIC.
6309.186:0128:0160:err:d3dcompiler:preprocess_shader     <anonymous>:11323:2: W4300: Redefinition of STATIC.
6309.186:0128:0160:err:d3dcompiler:preprocess_shader 
6309.198:0128:0160:err:d3dcompiler:D3DCompile2 Failed to compile shader, vkd3d result -4.
6309.198:0128:0160:err:d3dcompiler:D3DCompile2 Shader log:
6309.198:0128:0160:err:d3dcompiler:D3DCompile2     <anonymous>:270:1: W4300: Redefinition of varying_uvSet0_source.
6309.198:0128:0160:err:d3dcompiler:D3DCompile2     <anonymous>:275:1: W4300: Redefinition of FUNC_UVSET_1.
6309.198:0128:0160:err:d3dcompiler:D3DCompile2     <anonymous>:1040:2: W4300: Redefinition of STATIC.
6309.198:0128:0160:err:d3dcompiler:D3DCompile2     <anonymous>:11323:2: W4300: Redefinition of STATIC.
6309.198:0128:0160:err:d3dcompiler:D3DCompile2     <anonymous>:1395:8: E5005: Function "tex2Dlod" is not defined.
6314.632:0128:0170:err:sync:RtlpWaitForCriticalSection section 0440A2C8 (null) wait timed out in thread 0170, blocked by 0164, retrying (60 sec)
```

Specs
- OS: Arch
- Kernel: 6.6.52
- GPU: RX 6700 XT (Mesa 24.2.3)
- CPU: Ryzen 7 5700G (8c/16t)

**Comment by waspennator** [comment]
Seems like it got patched, retested with proton 10 and the water isn't going all rainbow seizure colors anymore
